File: examples/io_grpc/bazel_build_java/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/services/WhateverService.java
package com.geekinasuit.daggergrpc.iogrpc.example.armeria.services;

import com.geekinasuit.daggergrpc.api.GrpcCallScope;
import com.geekinasuit.daggergrpc.api.GrpcServiceHandler;
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.Whatever.WhateverRequest;
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.Whatever.WhateverResponse;
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.WhateverServiceGrpc;
import io.grpc.stub.StreamObserver;
import javax.inject.Inject;

@GrpcServiceHandler(grpcWrapperType = WhateverServiceGrpc.class)
@GrpcCallScope
public class WhateverService implements WhateverServiceGrpc.AsyncService {
  @Inject
  WhateverService() {}

  @Override
  public void whatever(WhateverRequest req, StreamObserver<WhateverResponse> resp) {
    WhateverResponse response =
        WhateverResponse.newBuilder().setSuceeded(req.getWhatever()).build();
    resp.onNext(response);
    resp.onCompleted();
  }
}

File: io_grpc/example/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/WhateverService.kt
package com.geekinasuit.daggergrpc.iogrpc.example.armeria

import com.geekinasuit.daggergrpc.api.GrpcCallScope
import com.geekinasuit.daggergrpc.api.GrpcServiceHandler
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.Whatever.WhateverRequest
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.Whatever.WhateverResponse
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.WhateverServiceGrpc
import io.grpc.stub.StreamObserver
import javax.inject.Inject

@GrpcServiceHandler(WhateverServiceGrpc::class)
@GrpcCallScope
class WhateverService @Inject constructor(): WhateverServiceGrpc.AsyncService {
  override fun whatever(
    request: WhateverRequest,
    responseObserver: StreamObserver<WhateverResponse>
  ) {
    val response = WhateverResponse.newBuilder().setSuceeded(request.whatever).build()
    responseObserver.onNext(response)
    responseObserver.onCompleted()
  }
}
